x/programs/runtime/import_program.go

```diff
@@ -35,12 +37,12 @@ func NewProgramModule(r *WasmRuntime) *ImportModule {
 	return &ImportModule{
 		Name: "program",
 		HostFunctions: map[string]HostFunction{
-			"call_program": {FuelCost: callProgramCost, Function: Function[callProgramInput, Result[RawBytes, uint8]](func(callInfo *CallInfo, input callProgramInput) (Result[RawBytes, uint8], error) {
+			"call_program": {FuelCost: callProgramCost, Function: Function[callProgramInput, Result[RawBytes, ErrorCode]](func(callInfo *CallInfo, input callProgramInput) (Result[RawBytes, ErrorCode], error) {
 				newInfo := *callInfo
 
 				// make sure there is enough fuel in current store to give to the new call
 				if callInfo.RemainingFuel() < input.Fuel {
-					return Err([]byte{byte(ExecutionFailure)}), nil
+					return Err[RawBytes, ErrorCode](OutOfFuel), nil
```

Comment:
In this case we wanted to just return an `ExecutionFailure`, since it should be the client responsibility to make sure that they have enough gas to complete the call.

Reply:
solved